extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -68,121 +86,141 @@ export class Messaging {
       event.origin != this.targetOrigin_) {
       return;
     }
+    /** @type {Message} */
     const message = event.data;
-    if (message.sentinel == requestSentinel_) {
+    if (message.app != APP) {
+      return;
+    }
+    if (message.type == MessageType_.REQUEST) {
       this.handleRequest_(message);
-    } else if (message.sentinel == responseSentinel_) {
+    } else if (message.type == MessageType_.RESPONSE) {
       this.handleResponse_(message);
-    } else {
-      throw new Error('Invalid Format!');
     }
   }
 
   /**
    * I'm sending Bob a new outgoing request.
    * @param {string} eventType
-   * @param {*} payload
+   * @param {*} data
    * @param {boolean} awaitResponse
    * @return {!Promise<*>|undefined}
    */
-  sendRequest(eventType, payload, awaitResponse) {
+  sendRequest(eventType, data, awaitResponse) {
     dev().info(TAG, 'messaging.js -> sendRequest, eventType: ', eventType);
-    const requestId = String(++this.requestIdCounter_);
+    const requestId = ++this.requestIdCounter_;
     let promise = undefined;
     if (awaitResponse) {
       promise = new Promise((resolve, reject) => {
         this.waitingForResponse_[requestId] = {resolve, reject};
       });
     }
-    this.sendMessage_(requestSentinel_, requestId, eventType,
-      payload, awaitResponse);
+    this.sendMessage_(requestId, MessageType_.REQUEST, eventType, data,
+      awaitResponse, null);
```

Comment:
Please annotate the `null`. Alternatively, since there are so many parameters, why not just do:
```
this.sendMessage_({
  a: b,
  c: d,
});
```

This gives you explicit naming everywhere and removes unnecessary `rsvp: undefined, error: undefined`.